lib/generators/data_migration/data_migration_generator.rb

```diff
@@ -15,10 +15,30 @@ class DataMigrationGenerator < Rails::Generators::NamedBase
       def create_data_migration
         set_local_assigns!
         migration_template template_path, data_migrations_file_path
+        create_data_migration_test
       end
 
       protected
 
+      def create_data_migration_test
+        return unless DataMigrate.config.test_support_enabled
+
+        case DataMigrate::Helpers::InferTestSuiteType.new.call
```

Comment:
nitpick:

... adding to my previous comment: would be nice if we could always rely on config here, instead of manually infering things each time